crates/core/src/db/datastore/traits.rs

```diff
@@ -419,7 +420,10 @@ pub trait MutTxDatastore: TxDatastore + MutTx {
     }
 
     // Indexes
-    fn create_index_mut_tx(&self, tx: &mut Self::MutTx, table_id: TableId, index: RawIndexDefV8) -> Result<IndexId>;
+
+    /// Create a new index with the given schema.
+    /// The index's ID field should be set to 0 (the autoinc sentinel value), but the table ID should not.
+    fn create_index_mut_tx(&self, tx: &mut Self::MutTx, index_schema: IndexSchema, is_unique: bool) -> Result<IndexId>;
```

Comment:
Why is `is_unique` necessary here? Also should it be included in the `index_schema` or no?

Reply:
See the comment added in `mut_tx.rs`. (It's a little hard to document these APIs because there's multiple layers with nearly-identical structures, not always clear where to put the comments.)

It is still necessary because the `table` crate still takes an `is_unique` parameter. We can eventually move the enforcement of unique constraints outside `table` and then it won't be necessary anymore.

It would be better to just look up the corresponding unique constraint in the system tables, but this might be being called in a transaction where the relevant unique constraint will be created later (in particular during `create_table`), so we need to hedge our bets and give the caller control.